### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by mcstudor**
[Build Status](https://dev.azure.com/mcstudor/CSCD379-2020-Winter/_build/latest?definitionId=1&branchName=Assignment4)

### CODE REVIEW

**Comment by mfirmin1** [comment]
I will Review this

**Comment by mfirmin1** [comment]
Assignment
-WebAPI Analyzers enabled ✔
-DI properly configured
Services properly registered with correct scopes ✔
-DB context properly registered ✔
    -AutoMapper properly registered ✔
    -NSwag properly setup ✔
    -API controllers properly expose CRUD operations ✔
- Controllers are all unit tested ✔
Extra Credit
-Unit tests in EntityServiceTests fixed approprietly ✔
-Mocking framework used to create the test doubles ✔

**Comment by mfirmin1** [comment]
Good job

File: SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        public async Task PutGift_Accepted_ReturnsGift()
+        {
+            Gift gift = SampleData.CreateChickenGift();
+            int giftId = 42;
+            _MockGiftService.Setup(svs => svs.UpdateAsync(42, It.IsAny<Gift>()))
+                .ReturnsAsync(gift);
+
+            using (var controller = new GiftController(_MockGiftService.Object))
+            {
+                ;
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs]: Stray semicolon?

File: SecretSanta/src/SecretSanta.Business/AutomapperConfigurationProfile.cs [hunk 2]
```diff
@@ -8,8 +8,13 @@ public class AutomapperConfigurationProfile : Profile
...
         {
-            CreateMap<Gift, Gift>().ForMember(property => property.Id, option => option.Ignore());
-            CreateMap<User, User>().ForMember(property => property.Id, option => option.Ignore());
+            CreateMap<EntityBase, EntityBase>()
+                .ForMember(property => property.Id, option => option.Ignore());
+            CreateMap<FingerPrintEntityBase, FingerPrintEntityBase>()
+                .ForMember(property => property.CreatedBy, option => option.Ignore())
+                .IncludeBase<EntityBase, EntityBase>();
+            CreateMap<Gift, Gift>().IncludeBase<FingerPrintEntityBase, FingerPrintEntityBase>();
+            CreateMap<User, User>().IncludeBase<FingerPrintEntityBase, FingerPrintEntityBase>();
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Business/AutomapperConfigurationProfile.cs]: Though this solution certainly works to make the unit tests happy, it does expose some problems. For example it still allows someone outside of the data layer to directly modify the finger printing properties (which really should be completely controlled in the data layer). You can see a solution in the Assignment4Complete branch.

**Pull request closed without merging**